src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImpl.java

```diff
@@ -5,55 +5,159 @@
     }
 
     @Override
-    public void onNodeMouseDoubleClick(NodeMouseDoubleClickEvent event) {
-        m_control.addControlPoint(event.getX(), event.getY());
+    public void onNodeMouseDoubleClick(final NodeMouseDoubleClickEvent event) {
+        doubleClickEventConsumer.accept(event);
     }
 
     public WiresConnectorControl getControl() {
-        return m_control;
+        return m_connector.getControl();
+    }
+
+    WiresConnector getConnector() {
+        return m_connector;
+    }
+
+    WiresManager getWiresManager() {
+        return m_wiresManager;
+    }
+
+    public static class DefaultClickHandlers {
+
+        private final Timer clickTimer;
+        private final WiresConnector connector;
+        private final boolean switchVisiblityOnClick;
+        private boolean isShiftKeyDown;
+
+        public DefaultClickHandlers(final WiresManager wiresManager,
+                                    final WiresConnector connector) {
+            this(connector,
+                 wiresManager.getSelectionManager(),
+                 true);
+        }
+
+        public DefaultClickHandlers(final WiresManager wiresManager,
+                                    final WiresConnector connector,
+                                    final boolean switchVisiblityOnClick) {
+            this(connector,
+                 wiresManager.getSelectionManager(),
+                 switchVisiblityOnClick);
+        }
+
+        private DefaultClickHandlers(final WiresConnector connector,
+                                    final SelectionManager selectionManager,
+                                     final boolean switchVisiblityOnClick) {
+            this.connector = connector;
+            this.switchVisiblityOnClick = switchVisiblityOnClick;
+            this.clickTimer = new Timer() {
+                @Override
+                public void run()
+                {
+                    if (selectionManager != null) {
+                        selectionManager.selected(connector, isShiftKeyDown);
+                    }
+                    if (switchVisiblityOnClick) {
+                        final WiresConnectorControl control = connector.getControl();
+                        if (control.areControlPointsVisible()) {
+                            control.hideControlPoints();
+                        } else {
+                            control.showControlPoints();
+                        }
+                    }
+                }
+            };
+        }
+
+        public Consumer<NodeMouseClickEvent> onClick() {
+            return new Consumer<NodeMouseClickEvent>() {
+                @Override
+                public void accept(NodeMouseClickEvent event) {
+                    cancelTimerIfRunning();
+                    isShiftKeyDown = event.isShiftKeyDown();
+                    clickTimer.schedule(150);
+                }
+            };
+        }
+
+        public Consumer<NodeMouseDoubleClickEvent> onDoubleClick() {
```

Comment:
the same here, the name could be something that is done by the method... for instance `addControlPoint` with parameter as `x` and `y`.